include/vcpkg/base/expected.h

```diff
@@ -368,7 +368,7 @@ namespace vcpkg
             ExpectedHolder<T> m_t;
         };
 
-        bool value_is_error;
+        bool value_is_error{false};
```

Comment:
I do not think this change is an improvement: the point is that each constructor needs to be very explicit with which alternative it is making active.